distributedcombigrid/src/sgpp/distributedcombigrid/manager/ProcessGroupWorker.cpp

```diff
@@ -751,6 +762,53 @@ void ProcessGroupWorker::evalErrorOnDFG() {
   sendEvalNorms(dfg);
 }
 
+std::vector<CombiDataType> ProcessGroupWorker::interpolateValues() {
+  assert(combiParameters_.getNumGrids() == 1 && "interpolate only implemented for 1 species!");
+  // receive coordinates and broadcast to group members
+  std::vector<real> interpolationCoordsSerial;
+  auto realType = abstraction::getMPIDatatype(abstraction::getabstractionDataType<real>());
+  int coordsSize;
+  MASTER_EXCLUSIVE_SECTION {
+    MPI_Status status;
+    MPI_Probe(theMPISystem()->getManagerRank(), TRANSFER_INTERPOLATION_TAG, theMPISystem()->getGlobalComm(), &status);
+    MPI_Get_count(&status, realType, &coordsSize);
+
+    // resize buffer to appropriate size and receive
+    interpolationCoordsSerial.resize(coordsSize);
+    MPI_Recv(interpolationCoordsSerial.data(), coordsSize, realType, theMPISystem()->getManagerRank(),
+             TRANSFER_INTERPOLATION_TAG, theMPISystem()->getGlobalComm(), MPI_STATUS_IGNORE);
+  }
+  // broadcast size of vector, and vector
+  MPI_Bcast(&coordsSize, 1, MPI_INT, theMPISystem()->getMasterRank(), theMPISystem()->getLocalComm());
+  interpolationCoordsSerial.resize(coordsSize);
+  MPI_Bcast(interpolationCoordsSerial.data(), coordsSize, realType,
+            theMPISystem()->getMasterRank(), theMPISystem()->getLocalComm());
+
+  // split vector into coordinates
+  const int dim = static_cast<int>(combiParameters_.getDim());
+  auto numCoordinates = coordsSize / dim;
+  assert( coordsSize % dim == 0);
+  std::vector<std::vector<real>> interpolationCoords(numCoordinates);
+  auto it = interpolationCoordsSerial.begin();
+  for (auto& coord: interpolationCoords) {
+    coord.insert(coord.end(), it, it+dim);
+    it += dim;
+  }
+  assert(it == interpolationCoordsSerial.end());
+
+  // call interpolation function on tasks and reduce with combination coefficient
+  std::vector<CombiDataType> values(numCoordinates, 0.);
+  for (Task* t : tasks_){
+    auto coeff = combiParameters_.getCoeff(t->getID());
+    auto taskVals = t->getDistributedFullGrid().getInterpolatedValues(interpolationCoords);
+
+    for (size_t i = 0; i < numCoordinates; ++i) {
+      values[i] += taskVals[i] * coeff;
+    }
+  }
+  return values;
```

Comment:
Don't you need to reduce the "values" across the process group in the end? Otherwise the master of the group does not have all contributions when he sends back the values.

Reply:
that should be handled by `getInterpolatedValues`, it contains an allreduce operation over the dfg.
(also, the test is passing in the current state, so it should be OK).

Reply:
oh OK, the tests are not passing on travis!
this here might be a hint: https://travis-ci.org/github/SGpp/DisCoTec/builds/763379629#L16442